substrate/frame/src/lib.rs

```diff
@@ -0,0 +1,449 @@
+//!
+//! # FRAME
+//!
+//! ```co_compile
+//!   ______   ______    ________   ___ __ __   ______
+//!  /_____/\ /_____/\  /_______/\ /__//_//_/\ /_____/\
+//!  \::::_\/_\:::_ \ \ \::: _  \ \\::\| \| \ \\::::_\/_
+//!   \:\/___/\\:(_) ) )_\::(_)  \ \\:.      \ \\:\/___/\
+//!    \:::._\/ \: __ `\ \\:: __  \ \\:.\-/\  \ \\::___\/_
+//!     \:\ \    \ \ `\ \ \\:.\ \  \ \\. \  \  \ \\:\____/\
+//!      \_\/     \_\/ \_\/ \__\/\__\/ \__\/ \__\/ \_____\/
+//! ```
+//!
+//! > **F**ramework for **R**untime **A**ggregation of **M**odularized **E**ntities: Substrate's
+//! > State Transition Function (Runtime) Framework.
+//!
+//! ## Warning: Experimental
+//!
+//! This crate and all of its content is experimental, and should not yet be used in production.
+//!
+//! ## Introduction
+//!
+//! Substrate is at a very high level composed of two parts:
+//!
+//! 1. A *runtime* which represents the state transition function (i.e. "Business Logic") of a
+//! blockchain, and is encoded as a Wasm blob.
+//! 2. A client whose primary purpose is to execute the given runtime.
+//!
+//! FRAME is the Substrate's framework of choice to build a runtime.
+//!
+//! FRAME is composed of two major components, **pallets** and a **runtime**. Pallets are isolated
+//! pieces of logic that can be composed together to form a runtime. Recall from the [`substrate`]'s
+//! architecture that a FRAME-based runtime would be part of a substrate client.
+#![doc = simple_mermaid::mermaid!("../../../docs/mermaid/substrate_with_frame.mmd")]
+//! ## Pallets
+//!
+//! A pallet is analogous to a module in the runtime, which can itself be composed of multiple
+//! components, most notable of which are:
+//!
+//! - Storage
+//! - Dispatchables
+//! - Events
+//! - Errors
+//!
+//! Most of these components are defined using macros, the full list of which can be found in
+//! [`frame_support::pallet_macros`]
+//!
+//! ###
+//!
+//! The following examples showcases a minimal pallet.
+#![doc = docify::embed!("src/lib.rs", pallet)]
+//!
+//! ## Runtime
+//!
+//! This crate also provides the ability to amalgamate multiple pallets into a single runtime. See
+//! [`runtime`]. To do so, the `runtime` feature of this crate needs to be enabled.
+//!
+//! ### Example
+//!
+//! The following example shows a (test) runtime that is composing the pallet demonstrated above,
+//! next to the [`frame_system`] pallet, into a runtime.
+#![doc = docify::embed!("src/lib.rs", runtime)]
+//!
+//! ## Underlying dependencies
+//!
+//! This crate is an amalgamation of multiple other crates that are often used together to compose a
+//! pallet. It is not necessary to use it, and it may fall short for certain purposes.
+//!
+//! In short, this crate only re-exports types and traits from multiple sources. All of these
+//! sources are listed (and re-exported again) in [`deps`].
+//!
+//! [`substrate`]: ../substrate/index.html
+
+#![cfg_attr(not(feature = "std"), no_std)]
+#![cfg(feature = "experimental")]
```

Comment:
Do we already build the docs with `experimental` feature?

Reply:
We do, see https://github.com/paritytech/polkadot-sdk/blob/447e7533237d038d43b2de68ff39e0a179503ac4/.gitlab/pipeline/build.yml#L103
and https://github.com/paritytech/polkadot-sdk/pull/1337/files#diff-4c9797095a391f4cc2b62de36ced891f3373d27d8ddb80557a07caf7af503eb8R11-R13